docs/source/reference/records.rst

```diff
@@ -21,7 +20,7 @@ Lexical structure
 
 * ``.foo.bar.baz`` is three lexemes: ``.foo``, ``.bar``, ``.baz``
 
-* If you want ``Constructor.field``, you have to write ``(Constructor).field``.
+* If you want ``Constructor.proj``, you have to write ``(Constructor).proj``.
```

Comment:
Is this `Constructor` a `TyCon` (i.e. you're using the record's namespace)?
Could we have a small example with a concrete record?

Reply:
At lexing/syntax level, we do not (cannot) distinguish between namespaces, data constructors and type constructors. All the lexer sees is a "capitalised identifier, dot, uncapitalised identifier", and lexes it as a namespaced name. If you want something like `MkFoo.bar`, where `MkFoo` is a (syntactically) nullary data constructor, you need to write `(MkFoo).bar` because a capitalised identifier _not_ followed by a dot is lexed as an (unqualified) identifier, not as a module name.

`(Constructor).field` is literally sugar for `field Constructor`; the desugarer does not look at whether `field` is record field, and does not qualify it using the record namespace. It leaves that to the disambiguator.

For example:
```idris
record Foo where
  constructor MkFoo
  {auto field : ()}

MkFoo.field  -- fails, parsed as NS ["MkFoo"] (UN "field")
(MkFoo).field  -- succeeds, parsed as (field MkFoo)
Foo.field  -- succeeds, parsed as NS ["Foo"] (UN "field")

MkFoo.Foo  -- fails, parsed as NS ["MkFoo"] (UN "Foo")
MkFoo.(Foo.field)  -- succeeds, parsed as (Foo.field MkFoo)
MkFoo.Foo.field  -- fails, parsed as NS ["Foo", "MkFoo"] (UN "field")
(MkFoo.Foo).field  -- fails, parsed as (field MkFoo.Foo)
```
Should I put this in the doc?

Reply:
> Should I put this in the doc?

I think so: a "nullary" constructor with (invisible) fields is exotic
enough that it would be easier to understand the original sentence
with an example.

Reply:
Perhaps `Z.printLn` would be a better example?

Reply:
I'll update it, thanks!

Reply:
(Or `True.not`.)

Reply:
Yes to all because these also demonstrate non-field examples! :D

Reply:
I'll overhaul the doc, actually; this should no longer be called "record projections", either.

Reply:
Beautiful!